src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,220 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int numberOfElements;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        numberOfElements = 0;
+        head = null;
+        tail = null;
+    }
+
     @Override
     public void add(T value) {
+        if (numberOfElements == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            newNode.previous = tail;
+            tail = newNode;
+        }
+        numberOfElements++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode;
+
+        if (index == numberOfElements) {
+            add(value);
+        } else {
+            newNode = new Node<>(null, value, null);
+            connectNodeBeforeIndex(newNode, index);
+            numberOfElements++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> result = giveNodeByIndex(index);
+        return result.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> oldNode = giveNodeByIndex(index);
+        Node<T> newNode = new Node<>(null, value, null);
+        changeConnectionFromToNode(oldNode, newNode);
+        return oldNode.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = giveNodeByIndex(index);
+        disconnectNode(node);
+        numberOfElements--;
+        return node.item;
     }
 
     @Override
     public boolean remove(T object) {
+        int count = 0;
+        Node<T> node;
+        while (count < numberOfElements) {
+            node = giveNodeByIndex(count);
+            if (node.item == null
+                    ? node.item == object : node.item.equals(object)) {
+                remove(count);
+                return true;
+            }
+            count++;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return numberOfElements;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return numberOfElements == 0;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("[ ");
+        Node<T> node;
+        int counter = 0;
+        while (counter < size()) {
+            node = giveNodeByIndex(counter);
+            stringBuilder.append(node);
+            if (counter + 1 < size()) {
+                stringBuilder.append(", ");
```

Comment:
I think " ," can be constant

Reply:
![image](https://user-images.githubusercontent.com/95774539/195266419-a854cc6f-ad7d-4dd2-bcf0-dcbd70ad047f.png)

It`s not make any sense. For example, toString() method from java libraries for all Collections